### PR TITLE
feat: add otel-collector sidecar

### PR DESCRIPTION
**Pull request by gburges**

### CODE REVIEW

File: controllers/cloud.redhat.com/providers/sidecar/provider.go [hunk 1]
```diff
@@ -4,7 +4,8 @@ import (
 	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers"
 )
 
-var DefaultImageSideCarTokenRefresher = "quay.io/observatorium/token-refresher:master-2023-09-20-f5e3403" // nolint:gosec
+var DefaultImageSideCarTokenRefresher = "quay.io/observatorium/token-refresher:master-2022-10-21-a99ce82" // nolint:gosec
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/sidecar/provider.go:7]: Take a look at the other implementations which use default images - we should have a way to override these in config, and even ClownEnv in some places

File: docs/site-readme.md [hunk 2]
```diff
@@ -1,154 +1 @@
...
-
-## Connect
-
-Any questions, please ask one of the Clowder development team
-
-* [@psav](https://github.com/psav)
-* [@bsquizz](https://github.com/bsquizz)
-* [@bennyturns](https://github.com/bennyturns)
-* [@adamrdrew](https://github.com/adamrdrew)
+docs/site-readme.md
```
Comment (psav) [inline docs/site-readme.md:1]: What is this change doing? - And why is it in this PR :)

Comment (gburges) [reply docs/site-readme.md]: i have no idea how that got there, restored the file

File: controllers/cloud.redhat.com/providers/sidecar/provider.go (hunk 1 above)
Comment (gburges) [reply controllers/cloud.redhat.com/providers/sidecar/provider.go]: added

File: controllers/cloud.redhat.com/providers/sidecar/default.go [hunk 3]
```diff
@@ -130,3 +144,25 @@ func getTokenRefresher(appName string) *core.Container {
 
 	return &cont
 }
+
+func getOtelCollector(appname string) *core.Container {
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/sidecar/default.go:148]: Lint issue here - appname is unused

File: apis/cloud.redhat.com/v1alpha1/clowdenvironment_types.go [hunk 4]
```diff
@@ -504,13 +504,30 @@ type ClowdEnvironmentSpec struct {
 type TokenRefresherConfig struct {
 	// Enables or disables token refresher sidecars
 	Enabled bool `json:"enabled"`
+	// Configurable image
+	Image TokenRefresherImage `json:"image,omitempty"`
```
Comment (psav) [inline apis/cloud.redhat.com/v1alpha1/clowdenvironment_types.go:508]: Is there a reason this needs a struct here? - The Image is just a string

File: controllers/cloud.redhat.com/providers/sidecar/provider.go [hunk 5]
```diff
@@ -1,10 +1,33 @@
...
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/clowderconfig"
 	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers"
 )
 
-var DefaultImageSideCarTokenRefresher = "quay.io/observatorium/token-refresher:master-2023-09-20-f5e3403" // nolint:gosec
+var DefaultImageSideCarTokenRefresher = "quay.io/observatorium/token-refresher:master-2022-10-21-a99ce82"                               // nolint:gosec
+var DefaultImageSideCarOtelCollector = "ghcr.io/os-observability/redhat-opentelemetry-collector/redhat-opentelemetry-collector:0.107.0" // nolint:gosec
+
+func GetFeatureFlagsUnleashImage(env *crd.ClowdEnvironment) string {
+	if env.Spec.Providers.Sidecars.TokenRefresher.Image.TokenRefresherImg != "" {
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/sidecar/provider.go:13]: Given my suggestion above,
this then would look like
```if env.Spec.Providers.Sidecars.TokenRefresher.Image```
Which reads better and is more succinct

File: tests/kuttl/test-sidecars/01-assert.yaml [hunk 6]
```diff
@@ -37,4 +38,5 @@ spec:
         spec:
           containers:
           - name: puptoo-cron
-          - name: token-refresher
\ No newline at end of file
+          - name: token-refresher
+          - name: otel-collector
```
Comment (psav) [inline tests/kuttl/test-sidecars/01-assert.yaml:42]: New line addition here please

File: tests/kuttl/test-sidecars/01-assert.yaml [hunk 7]
```diff
@@ -37,4 +38,5 @@ spec:
         spec:
           containers:
           - name: puptoo-cron
-          - name: token-refresher
\ No newline at end of file
+          - name: token-refresher
```
Comment (psav) [inline tests/kuttl/test-sidecars/01-assert.yaml:41]: I know it's an extra ask - but could we also have an assertion of the image here from the config? I'm not asking to test all versions, but just the default case.

**Review by psav — CHANGES_REQUESTED** [review]
Some changes please - apologies

File: controllers/cloud.redhat.com/providers/sidecar/default.go (hunk 3 above)
Comment (gburges) [reply controllers/cloud.redhat.com/providers/sidecar/default.go:148]: fixed

File: controllers/cloud.redhat.com/providers/sidecar/provider.go [hunk 8]
```diff
@@ -1,10 +1,33 @@
...
+	crd "github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/clowderconfig"
 	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers"
 )
 
-var DefaultImageSideCarTokenRefresher = "quay.io/observatorium/token-refresher:master-2023-09-20-f5e3403" // nolint:gosec
+var DefaultImageSideCarTokenRefresher = "quay.io/observatorium/token-refresher:master-2022-10-21-a99ce82"                               // nolint:gosec
+var DefaultImageSideCarOtelCollector = "ghcr.io/os-observability/redhat-opentelemetry-collector/redhat-opentelemetry-collector:0.107.0" // nolint:gosec
+
+func GetFeatureFlagsUnleashImage(env *crd.ClowdEnvironment) string {
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/sidecar/provider.go:12]: Function name is wrong here right?

File: docs/site-readme.md [hunk 9]
```diff
@@ -151,4 +151,4 @@ Any questions, please ask one of the Clowder development team
 * [@psav](https://github.com/psav)
 * [@bsquizz](https://github.com/bsquizz)
 * [@bennyturns](https://github.com/bennyturns)
-* [@adamrdrew](https://github.com/adamrdrew)
+* [@adamrdrew](https://github.com/adamrdrew)
```
Comment (psav) [inline docs/site-readme.md:154]: What's this change?

**Review by psav — CHANGES_REQUESTED** [review]
Small changes

File: docs/site-readme.md (hunk 9 above)
Comment (gburges) [reply docs/site-readme.md:154]: came with the make pre-push, doesn't do anything looks like

**Pull request merged**